scripts/cluster_network.py

```diff
@@ -173,6 +173,50 @@ def weighting_for_country(n, x):
     return (w * (100. / w.max())).clip(lower=1.).astype(int)
 
 
+def get_feature_for_hac(n, buses_i=None, feature=None):
+
+    if buses_i is None:
+        buses_i = n.buses.index
+
+    if feature is None:
+        feature = "solar+onwind-time"
+
+    carriers = feature.split('-')[0].split('+')
+    if "offwind" in carriers:
+        carriers.remove("offwind")
+        carriers = np.append(carriers, network.generators.carrier.filter(like='offwind').unique())
+
+    if feature.split('-')[1] == 'cap':
+        feature_data = pd.DataFrame(index=buses_i, columns=carriers)
+        for carrier in carriers:
+            try:
+                # without simpl wildcard (bus names are "X X"):
+                feature_data[carrier] = (n.generators_t.p_max_pu.filter(like=carrier).mean()
+                                         .rename(index=lambda x: x.split(' ')[0]))
+            except:
+                # with simpl wildcard (bus names are "X X X"):
+                feature_data[carrier] = (n.generators_t.p_max_pu.filter(like=carrier).mean()
+                                         .rename(index=lambda x: x.split(' ')[0] + ' ' + x.split(' ')[1]))
```

Comment:
I don't particularly like the try-except block. Can you pass the {simpl} wildcard as function argument and convert try-except to if-else?

Reply:
I think it can also be shortened. Maybe something like:

```py
if simpl:
    retrieve_func = lambda x: x.split(' ')[0] + ' ' + x.split(' ')[1]
else:
    retrieve_func = lambda x: x.split(' ')[0]

feature_data[carrier] = n.generators_t.p_max_pu.filter(like=carrier).mean().rename(index=retrieve_func)

Reply:
resolved with the suggestions of @coroa, because now no distinction is needed anymore.